Automattic/newspack-popups
# Shortcode (officially)

### All Submissions:






### Changes proposed in this Pull Request:



Adds ability to place a campaign anywhere, using a shortcode.
And a way to restrict a campaign to shortcode-only (manual).

### How to test the changes in this Pull Request:

1. Create an inline campaign, with "Every page" frequency.
2. Publish the campaign, observe that it's displayed on the frontend as expected
3. Add a shortcode – via shortcode block in a page, in a post, and in a text widget outside of the content area
4. Observe that the campaign is shown in all the shortcoded places
5. Observe that the campaign is not duplicated in the post content
6. In a different post (without the shortcode), observe the campaign is inserted automatically

### Caveats

If a campaign is placed via shortcode in a text widget, it will be treated as visible regardless of whether the widget area is actually rendered. This means that preventing duplication of campaigns in a widget area and in content is not feasible.
If there's a way to determine whether widget areas are rendered (regardless of the theme), this could be improved.

### Other information:
